### ISSUE TITLE
`litcli status` returns `running=true` for remote sub-servers that haven't been started

### ISSUE THREAD

**Issue by ViktorT-11** [lightninglabs/lightning-terminal#922]
When `litd` attempts to connect to a remote sub-server, the connection dialing is non-blocking in this function:  
https://github.com/lightninglabs/lightning-terminal/blob/601db231f722c8d9daab1aea8f86ff08539719b9/subservers/manager.go#L356-L382

As a result, if there is no server running at the specified RPC host, the function does not return an error because the connection status remains as "attempting to connect."

Since the sub-server manager directly sets a remote sub-server as `running=true` if the function does not return an error, the sub-server is marked as running here:  
https://github.com/lightninglabs/lightning-terminal/blob/601db231f722c8d9daab1aea8f86ff08539719b9/subservers/manager.go#L134-L141

**Comment by ellemouton** [comment]
@ViktorTigerstrom  - do you have a solution in mind & can this be assigned to you?

Perhaps some call-back on the first successful connection of some sort?

**Comment by ViktorTigerstrom** [comment]
Let's assign this to me :)!

> Perhaps some call-back on the first successful connection of some sort?

Yeah, i'll experiment a bit with that and determine best solution.

**Comment by Vandit1604** [comment]
@ViktorT-11 this overlaps my open #1373, so I want to flag it before the two drift apart.

#1373 adds a `watchRemoteConn` goroutine to `ConnectRemoteSubServers` in `subservers/manager.go`, right after the `SetRunning` you point at. That watcher tracks the remote connection state: it marks the sub-server errored on `TransientFailure` or `Shutdown`, and running again on `Ready`. It also nudges `Connect()` out of `Idle`, so an idle connection stays observable.

Because of that watcher, a remote sub-server that is never reachable no longer stays `running=true`. The dial in `dialBackend` (`manager.go:369`) is non-blocking, so it returns an `Idle` connection with no error, and we mark the sub-server running straight away. The watcher then drives that connection out of `Idle`, hits `TransientFailure`, and flips the status to errored. So the "healthy forever" symptom you describe here disappears once #1373 lands.

What #1373 does not fix is that first premature `SetRunning`. Removing it is the part you and @ellemouton discussed above: gate the initial running on the first real connection, not on the dial returning. That gate is a small extension of the same watcher — hold `SetRunning` until the connection first reaches `Ready`.

This issue has sat assigned to you since Jan 2025 with no PR, so I am happy to take the gating part. Do you want it folded into #1373, or a follow-up on top? And is first `Ready` the signal you had in mind, or something stricter?

**Comment by ViktorT-11** [comment]
> This issue has sat assigned to you since Jan 2025 with no PR, so I am happy to take the gating part. Do you want it folded into https://github.com/lightninglabs/lightning-terminal/pull/1373, or a follow-up on top? And is first Ready the signal you had in mind, or something stricter?

Thanks for calling this out @Vandit1604. From my viewpoint, I'd say that the smaller you can keep individual PRs which still actually adds functionality/addresses an issue, the better. So therefore I'd suggest doing it in a follow-up.

I'll re-assign this issue to you :)! Also, apologies for taking a long time to review your other PRs. We've been very busy with a lot of other high prio work to be done. But as soon as that is finished, I'll return to reviewing your work :pray:

**Comment by Vandit1604** [comment]
Thanks @ViktorT-11, and no worries about the timing at all.

A follow-up works for me. I'll build the gating on top of #1373 once it lands, so each PR stays small and self-contained. Whenever your high-priority work clears is fine — no rush on the reviews.
